render/present.go

```diff
@@ -0,0 +1,100 @@
+package render
+
+import (
+	"fmt"
+	"net/http"
+	"reflect"
+
+	"github.com/pressly/chi"
+
+	"golang.org/x/net/context"
+)
+
+var DefaultPresenter = NewPresenter()
+
+type Presenter interface {
+	Present(ctx context.Context, from interface{}) interface{}
+}
+
+// UsePresenter is a middleware that sets custom presenter into the context chain.
+func UsePresenter(p Presenter) func(next chi.Handler) chi.Handler {
+	return func(next chi.Handler) chi.Handler {
+		fn := func(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+			ctx = context.WithValue(ctx, presenterCtxKey, p)
+			next.ServeHTTPC(ctx, w, r)
+		}
+		return chi.HandlerFunc(fn)
+	}
+}
+
+func NewPresenter() *presenter {
+	return &presenter{
+		ConversionFnStore: map[reflect.Type]reflect.Value{},
+	}
+}
+
+type presenter struct {
+	ConversionFnStore map[reflect.Type]reflect.Value // map[*from.Type]func(context.Context, *from.Type) (*to.Type, error)
+}
+
+func (p *presenter) Register(conversionFunc interface{}, conversionFuncs ...interface{}) {
+	if err := p.register(conversionFunc); err != nil {
+		panic(err)
+	}
+	for _, fn := range conversionFuncs {
+		if err := p.register(fn); err != nil {
+			panic(err)
+		}
+	}
+}
+
+func (p *presenter) RegisterFrom(presenter *presenter, presenters ...*presenter) {
```

Comment:
This copies Presenter conversion functions from other presenters.. Should we call it .Copy()?